Organizely/src/app/classes-page/classes-page.component.ts

```diff
@@ -41,21 +41,51 @@ export class ClassesPageComponent implements OnInit {
       },
       (error) => {
         console.log(error);
-      }
+      },
+      () => console.log('All done getting your courses.')
     );
   }
 
   onDeleteCourse(id: number) {
     if (confirm('Are you sure you want to delete this course?')) {
       this.coursesService.deleteCourse(id).subscribe(
-        (response) => {
-          // TODO: Determine why this returns 'null'
-          console.log(response);
-          this.getAllCourses();
+        (data: void) => {
+          let coursesOfCurrentSem = this.semestersBySchoolYear[
+            this.currentSemester['semesterYear']
+          ][this.currentSemester['semesterSeason']];
+
+          let currentCourseIndex: number = this.currentCourses.findIndex(
+            (currentCourse) => currentCourse.courseId === id
+          );
+
+          let courseIndex: number = this.courses.findIndex(
+            (course) => course.courseId === id
+          );
+
+          let courseInSemIndex: number = coursesOfCurrentSem.findIndex(
+            (course) => course.courseId === id
+          );
+
+          this.semestersBySchoolYear[this.currentSemester['semesterYear']][
+            this.currentSemester['semesterSeason']
+          ].splice(courseInSemIndex, 1);
+          this.currentCourses.splice(currentCourseIndex, 1);
+          this.courses.splice(courseIndex, 1);
+
+          if (
+            this.semestersBySchoolYear[this.currentSemester['semesterYear']][
+              this.currentSemester['semesterSeason']
+            ].length === 0
+          ) {
+            delete this.semestersBySchoolYear[
+              this.currentSemester['semesterYear']
+            ];
+          }
```

Comment:
I don't understand what you did there, but if it works properly I approve it!

Reply:
The `delete` is a keyword used to delete a specific key from an object. This line of code deletes the key from the `semestersBySchoolYear` object (ex. 2021) if it has an empty array. This ensures the school year view displays the 'No Courses' message instead of blank space when there are no key-value pairs.